crates/extensions/tedge_downloader_ext/src/actor.rs

```diff
@@ -69,14 +69,16 @@ impl DownloaderActor {
 
 #[async_trait]
 impl Server for DownloaderActor {
-    type Request = DownloadRequest;
-    type Response = DownloadResult;
+    type Request = (String, DownloadRequest);
```

Comment:
Instead of this, why not make the `String` id a part of the `DownloadRequest` itself? Same for `DownloadResponse` as well.

Reply:
For the response, `(String, Result<DownloadResponse, DownloadError>)` is simpler because one also needs a request id for the errors. Doing the same for requests is then more natural.

Reply:
Another alternative would have been to change the `DownloadResult` from a type alias to an explicit struct with fields `id: String` and `result: Result<DownloadResponse, DownloadError>)`. But this is fine for now.